orc8r/cloud/go/services/metricsd/prometheus/prometheus-cache/cache/cache.go

```diff
@@ -147,18 +148,51 @@ func (c *MetricCache) clearMetrics() {
 	c.metricFamiliesByName = make(map[string]*familyAndMetrics)
 }
 
-func (c *MetricCache) exposeMetrics(metricFamiliesByName map[string]*familyAndMetrics) string {
-	respStr := strings.Builder{}
+func (c *MetricCache) exposeMetrics(metricFamiliesByName map[string]*familyAndMetrics, workers int) string {
+	fams := make(chan *familyAndMetrics, workers)
+	results := make(chan string, workers)
+	respStr := make(chan string, 1)
+
+	waitGroup := &sync.WaitGroup{}
+
+	for i := 0; i < workers; i++ {
+		waitGroup.Add(1)
+		go processFamilyWorker(fams, results, waitGroup)
+	}
+
+	go processFamilyStringsWorker(results, respStr)
+
 	for _, fam := range metricFamiliesByName {
+		fams <- fam
+	}
+
+	close(fams)
```

Comment:
Since `fams` is buffered, you could still have workers processing data when you close this channel. Close the channels after you are done waiting for the wg

Reply:
see above